mentoring/mentoring.py

```diff
@@ -83,54 +65,105 @@ class MentoringBlock(XBlockWithLightChildren, StepParentMixin):
-    followed_by = String(help="url_name of the step after the current mentoring block in workflow",
-                         default=None, scope=Scope.content)
-    url_name = String(help="Name of the current step, used for URL building",
-                      default='mentoring-default', scope=Scope.content)
-    enforce_dependency = Boolean(help="Should the next step be the current block to complete?",
-                                 default=False, scope=Scope.content, enforce_type=True)
+        return value == default_xml_content
+
+    # Content
+    MENTORING_MODES = ('standard', 'assessment')
+    mode = String(
+        help="Mode of the mentoring. 'standard' or 'assessment'",
+        default='standard',
+        scope=Scope.content,
+        values=MENTORING_MODES
+    )
+    followed_by = String(
+        help="url_name of the step after the current mentoring block in workflow.",
+        default=None,
+        scope=Scope.content
+    )
+    max_attempts = Integer(
+        help="Number of max attempts for this questions",
+        default=0,
+        scope=Scope.content,
+        enforce_type=True
+    )
+    url_name = String(
+        help="Name of the current step, used for URL building",
+        default='mentoring-default',  # TODO in future: set this to xblock.fields.UNIQUE_ID and remove self.url_name_with_default
+        scope=Scope.content
+    )
+    enforce_dependency = Boolean(
+        help="Should the next step be the current block to complete?",
+        default=False,
+        scope=Scope.content,
+        enforce_type=True
+    )
     display_submit = Boolean(help="Allow to submit current block?", default=True, scope=Scope.content)
-    xml_content = String(help="XML content", default=_default_xml_content, scope=Scope.content)
-    weight = Float(help="Defines the maximum total grade of the block.",
-                   default=1, scope=Scope.content, enforce_type=True)
-    num_attempts = Integer(help="Number of attempts a user has answered for this questions",
-                           default=0, scope=Scope.user_state, enforce_type=True)
-    max_attempts = Integer(help="Number of max attempts for this questions", default=0,
-                           scope=Scope.content, enforce_type=True)
-    mode = String(help="Mode of the mentoring. 'standard' or 'assessment'",
-                  default='standard', scope=Scope.content)
-    step = Integer(help="Keep track of the student assessment progress.",
-                   default=0, scope=Scope.user_state, enforce_type=True)
-    student_results = List(help="Store results of student choices.", default=[],
-                           scope=Scope.user_state)
-
-    display_name = String(help="Display name of the component", default="Mentoring XBlock",
-                          scope=Scope.settings)
+    xml_content = String(help="XML content", default=default_xml_content, scope=Scope.content)
```

Comment:
@Kelketek Unless I've missed something, the reason there has been multiple changes around url_name is because we have been struggling to find the right approach for initializing the field that would get the assent of upstream - there has been multiple iterations over this. But now that we have a good and agreed solution merged, and that we only have to wait for it to become available, so not sure if we want to go yet another way, with potentially a new set of issues - the workbench compatibility you were discussing being one example.

For the XBlock fork, you can see the difference here: https://github.com/edx-solutions/XBlock/commits/master there are only the FIELDS_TO_INIT and your reimplementation of child serialization, so we should likely be able to take the current XBlock master (not the one pointed by birch though, which is pointing to an older XBlock commit currently).